File: RGBCubeModel/src/de/jonas/cubeModel/animations/RainfallAnimation.java
package de.jonas.cubeModel.animations;

import de.jonas.cubeModel.Constants;
import de.jonas.cubeModel.Cube;

public class RainfallAnimation extends Animation {

	private int layerToLight = 7;
	private boolean multiplex;
	private int row, column;

	public RainfallAnimation(Cube cubes) {
		super(cubes);
		turnAllLayersOff();

		turnLayerOn(7);
		setAllLedsWhite();
	}

	@Override
	public void animate1ms() {
		if (multiplex) {
			clearAllLeds();
			setLedWhite(row, column);
			turnLayerOff(layerToLight + 1);
			turnLayerOn(layerToLight);
		} else {
			setAllLedsWhite();
			turnLayerOff(layerToLight);
			turnLayerOn(7);
		}
		multiplex = !multiplex;

	}

	@Override
	public void animate50ms() {

	}

	@Override
	public void animate100ms() {
		if (layerToLight == 0) {
			turnLayerOff(layerToLight + 1);
			layerToLight = rand.nextInt(15) + 10;
			row = rand.nextInt(Constants.numberOfLedsInRow);
			column = rand.nextInt(Constants.numberOfLedsInColumn);
		}
		layerToLight--;

	}

	@Override
	public void animate200ms() {

	}

	@Override
	public void animate500ms() {
		// TODO Auto-generated method stub

	}

	@Override
	public void animate1000ms() {
		// TODO Auto-generated method stub

	}

	@Override
	public String getAnimationName() {
		return "Regenschauer";
	}

	@Override
	public String getAudioFileName() {
		return "bla.mp3";
	}

	@Override
	public void reset() {
		milliSecondsToRun = INFINITY;
		turnAllLayersOff();

		turnLayerOn(7);
		setAllLedsWhite();
		
	}

	@Override
	void loop() {
		// TODO Auto-generated method stub
		
	}

}

File: RGBCubeControlKotlin/app/src/main/java/de/jonas/rgbcubecontrol/domain/animations/RainfallAnimation.kt
package de.jonas.rgbcubecontrol.domain.animations

import de.jonas.rgbcubecontrol.domain.Constants

class RainfallAnimation : Animation() {

    private var layerToLight = 7
    private var multiplex: Boolean = false
    private var row: Int = 0
    private var column: Int = 0

    override val animationName: String
        get() = "Rainfall"

    init {
        turnAllLayersOff()

        turnLayerOn(7)
        setAllLedsWhite()
    }

    override fun animate2ms() {
        if (multiplex) {
            clearAllLeds()
            setLedWhite(row, column)
            turnLayerOff(layerToLight + 1)
            turnLayerOn(layerToLight)
        } else {
            setAllLedsWhite()
            turnLayerOff(layerToLight)
            turnLayerOn(7)
        }
        multiplex = !multiplex

    }



   override fun animate100ms() {
        if (layerToLight == 0) {
            turnLayerOff(layerToLight + 1)
            layerToLight = rand.nextInt(15) + 10
            row = rand.nextInt(Constants.numberOfLedsInRow)
            column = rand.nextInt(Constants.numberOfLedsInColumn)
        }
        layerToLight--

    }



    override fun reset() {
        turnAllLayersOff()

        turnLayerOn(7)
        setAllLedsWhite()

    }
}